File: app/src/main/java/com/michelbarbosa/hsdm_hearthstonedustmanager/utils/MapperUtil.java
package com.michelbarbosa.hsdm_hearthstonedustmanager.utils;

import com.michelbarbosa.hsdm_hearthstonedustmanager.data.domain.SetWeight;
import com.michelbarbosa.hsdm_hearthstonedustmanager.data.domain.Stereotype;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class MapperUtil {

    public static List<Stereotype> getDefaultStereotype(String... defaultList) {
        List<String> stereotypeNames = new ArrayList<>(Arrays.asList(defaultList));
        List<Stereotype> stereotypeList = new ArrayList<>();
        int i = 0;
        for (String item : stereotypeNames) {
            stereotypeList.add(new Stereotype(i, item));
            i++;
        }
        return stereotypeList;
    }

    public static List<SetWeight> getDefaultSetWeight(int defaultValueAll, String... defaultList){
        List<String> setWeightNames = new ArrayList<>(Arrays.asList(defaultList));
        List<SetWeight> setWeightList = new ArrayList<>();
        int i = 0;
        for (String item: setWeightNames) {
            setWeightList.add(new SetWeight(i, item, defaultValueAll));
            i++;
        }
        return setWeightList;
    }

}

File: app/src/main/java/com/michelbarbosa/hsdm_hearthstonedustmanager/utils/MapperUtil.kt
package com.michelbarbosa.hsdm_hearthstonedustmanager.utils

import com.michelbarbosa.hsdm_hearthstonedustmanager.data.domain.SetWeight
import com.michelbarbosa.hsdm_hearthstonedustmanager.data.domain.Stereotype
import java.util.*

object MapperUtil {
    fun getDefaultStereotype(vararg defaultList: String?): List<Stereotype> {
        val stereotypeNames: List<String> = ArrayList(Arrays.asList(*defaultList))
        val stereotypeList: MutableList<Stereotype> = ArrayList()
        var i = 0
        for (item in stereotypeNames) {
            stereotypeList.add(Stereotype(i, item))
            i++
        }
        return stereotypeList
    }

    fun getDefaultSetWeight(defaultValueAll: Int, vararg defaultList: String?): List<SetWeight> {
        val setWeightNames: List<String> = ArrayList(Arrays.asList(*defaultList))
        val setWeightList: MutableList<SetWeight> = ArrayList()
        var i = 0
        for (item in setWeightNames) {
            setWeightList.add(SetWeight(i, item, defaultValueAll))
            i++
        }
        return setWeightList
    }
}